<launch>
    <!-- realsense -->

    <!-- my_img_proc -->
    <node name="fb_control_launch" pkg="img_proc" type="fb_control" required="true" output="screen">
        <rosparam command="load" file="$(find img_proc)/config/param.yaml" />
    </node>

    <!-- dobot control -->
    <node name="dobot_control_sub" pkg="img_proc" type="dobot_control_sub" required="true" output="screen"></node>

    <!-- dynamixel_workbench_controllers: dynamixel_controllers.launch -->
    <arg name="usb_port"                default="/dev/ttyUSB0"/>
    <arg name="dxl_baud_rate"           default="1000000"/>
    <arg name="namespace"               default="dynamixel_workbench"/>
    <arg name="use_moveit"              default="false"/>
    <arg name="use_joint_state"         default="true"/>
    <arg name="use_cmd_vel"             default="false"/>
    <param name="dynamixel_info"          value="$(find dynamixel_workbench_controllers)/config/basic.yaml"/>
    <node name="$(arg namespace)" pkg="dynamixel_workbench_controllers" type="dynamixel_workbench_controllers"
            required="true" output="screen" args="$(arg usb_port) $(arg dxl_baud_rate)">
        <param name="use_moveit"              value="$(arg use_moveit)"/>
        <param name="use_joint_states_topic"  value="$(arg use_joint_state)"/>
        <param name="use_cmd_vel_topic"       value="$(arg use_cmd_vel)"/>
        <rosparam>
            publish_period: 0.010
            dxl_read_period: 0.010
            dxl_write_period: 0.010
            mobile_robot_config:   
                seperation_between_wheels: 0.160  
                radius_of_wheel: 0.033            
        </rosparam>
    </node>

</launch>